samples/excel/89-preview-apis/protect-data-in-worksheet-and-workbook-structure.yaml

```diff
@@ -0,0 +1,258 @@
+id: excel-protect-data-in-worksheet-and-workbook-structure
+name: Protect data in worksheet and the workbook structure
+description: Shows how to protect data in a worksheet and the workbook structure.
+host: EXCEL
+api_set:
+    ExcelApi: 1.7
+script:
+    content: |
+        $("#setup").click(() => tryCatch(setup));
+        $("#protect-data-in-worksheet").click(() => tryCatch(protectDataInWorksheet));
+        $("#unprotect-data-in-worksheet").click(() => tryCatch(unprotectDataInWorksheet));
+        $("#protect-workbook-structure").click(() => tryCatch(protectWorkbookStructure));
+        $("#unprotect-workbook-structure").click(() => tryCatch(unprotectWorkbookStructure));
+        $("#password-protect-data-in-worksheet").click(() => tryCatch(passwordProtectDataInWorksheet));
+        $("#password-unprotect-data-in-worksheet").click(() => tryCatch(passwordUnprotectDataInWorksheet));
+        $("#password-protect-workbook-structure").click(() => tryCatch(passwordProtectWorkbookStructure));
+        $("#password-unprotect-workbook-structure").click(() => tryCatch(passwordUnprotectWorkbookStructure));
+
+        async function protectDataInWorksheet() {
+            await Excel.run(async (context) => {
+                await worksheetHelper(context, true, false);
+            });
+        }
+
+        async function unprotectDataInWorksheet() {
+            await Excel.run(async (context) => {
+                await worksheetHelper(context, false, false);
```

Comment:
One problem with "Helper" methods is that they are not self-documenting. (Help what?) And they often violate the principle that any method should do just one thing. Your helpers do multiple things, including opposite things; e.g. they protect and they unprotect. You depend on the parameters passed to them to determine what they do in any given call. If you keep the helpers, please add comments to each parameter explaining what effect it's going to have. But I would really recommend that you ditch the helper methods and have each top-level method do exactly what it's name says; e.g., "unprotectDataInWorksheet". You'll have a lot of repetitive code, but that's OK in snippets, even desirable.